server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -730,6 +734,12 @@ public Pool createAndEnrichPools(Pool pool, List<Pool> existingPools) {
             createPool(p);
         }
 
+        for (Pool tempPool : pools) {
+            if (pool.isLocked()) {
```

Comment:
This logic looks like it'll be fine, but we're at risk of losing the update to the locked state and we're doing a second iteration through the pools.

Conveniently, if we look through ```poolRules.createAndEnrichPools```, we get to a method called ```clonePool``` in ```PoolHelper```. I know I said before "the fewer places we do the locking the better," ```clonePool``` is one such place where we could use it. It also eliminates needing to set the locked flag here, which is somewhat random relative to the name of the method.

PoolHelper.java:255 right before the ```// Copy upstream fields``` comment would be a good place for the check + set logic used here.